test/observer_test.dart

```diff
@@ -88,94 +23,128 @@ void main() async {
-    final o = ObservableSingle.fromStore(store);
-    var putCount = 0;
-    o.observeSingleType(testEntityId, (Pointer<Void> user_data) {
-      putCount++;
-    }, randomPtr);
-
+  test('Observe single entity', () async {
+    Completer completer;
+    var expectedEvents = 0;
+
+    final stream = env.store.subscribe<TestEntity>();
+    final subscription = stream.listen((_) {
+      print('TestEntity updated');
+      expectedEvents--;
+      if (expectedEvents == 0) {
+        completer.complete();
+      }
+    });
+
+    // expect two events after one put() and one putMany()
+    expectedEvents = 2;
+    completer = Completer();
+    box.put(simpleStringItems[0]);
+    Box<TestEntity2>(env.store).put(TestEntity2());
     box.putMany(simpleStringItems);
-    simpleStringItems.forEach((i) => box.put(i));
-    simpleNumberItems.forEach((i) => box.put(i));
-
-    bindings.obx_observer_close(ObservableSingle.observer);
-    expect(putCount, 13);
+    await completer.future.timeout(defaultTimeout);
+    expect(expectedEvents, 0);
+
+    // cancel the subscription
+    await subscription.cancel();
+
+    // make sure there are no more events after cancelling
+    expectedEvents = 1;
+    completer = Completer();
+    box.put(simpleStringItems[0]);
+    expect(completer.future.timeout(defaultTimeout),
+        throwsA(isA<TimeoutException>()));
+    expect(expectedEvents, 1); // note: unchanged, no events received anymore
   });
 
-  test('Observe any entity with static callback', () async {
-    final callback = Pointer.fromFunction<obx_observer>(callbackAnyType);
-    final observer =
-        bindings.obx_observe(store.ptr, callback, Pointer.fromAddress(1337));
+  test('Observe multiple entities', () async {
+    Completer completer;
+    var expectedEvents = 0;
+    var typesUpdates = <Type, int>{}; // number of events per entity type
 
-    box.putMany(simpleStringItems);
+    final stream = env.store.subscribeAll();
 
-    box.remove(1);
+    final subscription = stream.listen((entityType) {
+      print('Entity updated: $entityType');
+      expectedEvents--;
 
-    // update value
-    final entity2 = box.get(2);
-    entity2.tString = 'Dva';
-    box.put(entity2);
+      if (typesUpdates[entityType] == null) {
+        typesUpdates[entityType] = 0;
+      }
+      typesUpdates[entityType]++;
 
-    final box2 = Box<TestEntity2>(store);
-    box2.put(TestEntity2());
-    box2.remove(1);
-    box2.put(TestEntity2());
-
-    expect(callbackAnyTypeCounter, 6);
-    bindings.obx_observer_close(observer);
-  });
-
-  test('Observe single entity', () async {
-    final callback =
-        Pointer.fromFunction<obx_observer_single_type>(callbackSingleType);
-    final observer = bindings.obx_observe_single_type(
-        store.ptr, testEntityId, callback, randomPtr);
+      if (expectedEvents == 0) {
+        completer.complete();
+      }
+    });
 
+    // expect two events after one put() and one putMany()
```

Comment:
Maybe here, and in above test as well change the note so sth. like "expect to receive events for both entity types" (and above e.g. "expect to receive events for only TestEntity").

Anyhow, should be "three events".